src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -78,7 +81,11 @@ private void processBlob(BlobClient blob) {
             .ifPresentOrElse(
                 envelope -> {
                     if (envelope.status == Status.CREATED) {
-                        leaseAndThen(blob, () -> blobProcessor.continueProcessing(envelope.id, blob));
+                        leaseAndThen(
+                            blob,
+                            this::isEnvelopeInCreatedStatus,
```

Comment:
Why are we involving this check in lease acquiring process?

Reply:
Also, blobProcessor.process will need to do some custom stuff, too:
- create the envelope and process the blob when there's no envelope
- pick up the existing envelope if it's in the created state
- abandon processing if the envelope is not in created state

Reply:
All this is done in LeaseAcquirer. And the check is called not while acquiring the lease, but after that.